src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/AccessTokenHttpMessageHandler.cs

```diff
@@ -19,13 +21,36 @@ public AccessTokenHttpMessageHandler(HttpMessageHandler inner, HttpConnection ht
 
     protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
     {
-        var accessToken = await _httpConnection.GetAccessTokenAsync().ConfigureAwait(false);
+        var isNegotiate = false;
+        if (string.IsNullOrEmpty(_accessToken) ||
+            // Negotiate redirects likely will have a new access token so let's always grab a (potentially) new access token on negotiate
+            request.RequestUri!.OriginalString.Contains("/negotiate?"))
```

Comment:
The only thing I would maybe change is a more definitive way to know it's the negotiation request by adding something to HttpRequestMessage.Properties/Options.

Reply:
HttpRequestMessage.Properties/Options allocates once accessed, not the worst thing as it's client-side, but worth noting. The only other way I can think of is to add a header during negotiate, but that's super ugly and bleeds into the network calls.